cumulusci/tasks/create_package_version.py

```diff
@@ -144,6 +144,11 @@ class CreatePackageVersion(BaseSalesforceApiTask):
     def _init_options(self, kwargs):
         super()._init_options(kwargs)
 
+        if not self.org_config.config_file:
+            raise TaskOptionsError(
+                "Org doesn't contain a config_file. It's a persistent org like Devhub or Developer Edition org"
```

Comment:
While this error message is better, I think we can still improve it. In general, we want to explain what happened and why, and help the user fix it. With that in mind, how about something like this?:
```suggestion
                "Target org scratch org definition file missing. Persistent orgs (e.g., Dev Hub) can't be used for 2GP package uploads."
```

In addition, the ACs for this story include task documentation. You can add`task_docs` to the class (see [this example](https://github.com/SFDO-Tooling/CumulusCI/blob/210f78cccb744b7cc9a1fbf19c2cf50805c0f47c/cumulusci/tasks/bulkdata/generate_mapping.py#L14-L34)) to let users know what the org is for when they run `cci task info`. Something like:
```python
task_docs = """
    Facilitates the upload of 2GP (second-generation packaging)
    package versions using CumulusCI.

    The target org is used both for looking up dependency package IDs and
    configuring the build org during the package upload. Ensure the specified
    org is a scratch org with the correct configuration for these purposes.
    """
```